app/models/filter_tag.rb

```diff
@@ -8,12 +8,14 @@ class GlobalFilter::FilterTag < ::ActiveRecord::Base
   ]
 
   def self.categories_for_tags(tags, scope)
-    filter_tags = self.where(name: tags)
-    filter_tags_category_ids =
-      filter_tags
-        &.pluck(:category_ids)
-        &.flat_map { |c| c.present? ? c.split("|") : Category.secured(scope).pluck(:id) }
+    @cached_categories ||= begin
```

Comment:
This does not seem correct. This is a class method which accepts arguments and we are using a class instance variable to memoize the return value. Once this method has been called once, it will always return the memoized value even if the arguments to the method is different.

Reply:
It is most definitely not correct 👍